starter/src/main/java/io/javaoperatorsdk/operator/springboot/starter/CRDApplier.java

```diff
@@ -58,23 +56,21 @@ public DefaultCRDApplier(KubernetesClient kubernetesClient, List<CRDTransformer>
     @Override
     public void apply() {
       log.debug("Uploading CRDs with suffix {} under {}", crdSuffix, crdPath);
-      stream(findFiles()).forEach(this::applyCrd);
+      stream(findResources()).forEach(this::applyCrd);
     }
 
-    private File[] findFiles() {
-      var resource = requireNonNull(
-          getClass().getResource(crdPath),
-          "Could not find the configured CRD path");
-
+    private Resource[] findResources() {
+      final var resourceResolver = new PathMatchingResourcePatternResolver();
+      final var resourceLocationPattern = crdPath + '*' + crdSuffix;
```

Comment:
Users may find it more useful if they can specify the entire location pattern or even a set of patterns (e.g., `META-INF/fabric8/*-v1.yml` and `META-INF/fabric8/*-v2.yml`). This is a breaking change and outside the scope of the fix, so let me leave it for now.